pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -426,4 +495,175 @@ describe('TimelockAuthorizer execute', () => {
+      // which it must have in order to be able to schedule calls to it.
+      const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+      await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, 2 * delay, { from: root });
+    });
+
+    sharedBeforeEach('grant permission', async () => {
+      const action = await actionId(authenticatedContract, 'protectedFunction');
+      await authorizer.grantPermission(action, user, authenticatedContract, { from: root });
+
+      await authorizer.scheduleAndExecuteDelayChange(action, delay, { from: root });
+    });
+
+    const schedule = async (functionData: string): Promise<number> => {
+      return authorizer.schedule(
+        authenticatedContract.address,
+        authenticatedContract.interface.encodeFunctionData('protectedFunction', [functionData]),
+        [],
+        { from: user }
+      );
+    };
+
+    // We will schedule 5 calls to `protectedFunction`, with data 0x00, 0x01, 0x02, etc.
+    const SCHEDULED_ENTRIES = 5;
+    // There'll be two extra entries: the one used to set the delay for setAuthorizer, and the one used to set the delay
+    // for `protectedFunction`.
+    const TOTAL_ENTRIES = SCHEDULED_ENTRIES + 2;
+
+    sharedBeforeEach('schedule multiple entries', async () => {
+      for (let i = 0; i < SCHEDULED_ENTRIES; ++i) {
+        await schedule(`0x0${i}`);
+      }
+    });
+
+    it('returns the number of total entries', async () => {
+      expect(await authorizer.instance.getScheduledExecutionsCount()).to.equal(TOTAL_ENTRIES);
+    });
+
+    it('reverts if the skip value is too large', async () => {
+      await expect(authorizer.instance.getScheduledExecutions(TOTAL_ENTRIES, 10, false)).to.be.revertedWith(
+        'SKIP_VALUE_TOO_LARGE'
+      );
+    });
+
+    it('reverts if the maxSize value is zero', async () => {
+      await expect(authorizer.instance.getScheduledExecutions(0, 0, false)).to.be.revertedWith('ZERO_MAX_SIZE_VALUE');
+    });
+
+    context('in chronological order', () => {
+      const reverseOrder = false;
+
+      it('returns entries in chronological order', async () => {
+        const entries = await authorizer.instance.getScheduledExecutions(0, 10, reverseOrder);
+
+        expect(entries.length).to.equal(TOTAL_ENTRIES);
+
+        // The first entry is the one that sets the setAuthorizer delay
+        expect(entries[0].where).to.equal(authorizer.address);
+        expect(entries[0].data).to.equal(
+          authorizer.interface.encodeFunctionData('setDelay', [await actionId(vault, 'setAuthorizer'), 2 * delay])
+        );
+
+        // The last entry is the one that we scheduled
+        expect(entries[entries.length - 1].where).to.equal(authenticatedContract.address);
+        expect(entries[entries.length - 1].data).to.equal(
+          authenticatedContract.interface.encodeFunctionData('protectedFunction', [`0x04`])
+        );
```

Comment:
Would it make sense to validate the ones in the middle with (or without) a loop?

Reply:
Eh I dunno, it'd be a bit annoying, and by looking at the code I think it's clear that if the edges are correct then everything else is as well :shrug: